rfcs/rdm-0064-custom-fields.md

```diff
@@ -0,0 +1,440 @@
+
+**create**
+
+You can add fields to the records and community data models using the following command. You can pass any number of configured fields.
+
+```shell
+invenio custom-fields create [-f <fieldname>] [-f <fieldname>]
+```
+
+In order to create _all_ configured values, just ignore the `-f` option. For example, on the first time an instance is set up.
+
+```shell
+invenio custom-fields create
+```
+
+### Adding a custom UI widget
+
+The previously mentioned `RDM_RECORDS_CUSTOM_FIELDS_UI` variable holds the information about how the custom fields will be displayed in the different UI pages.
+
+As it was mentioned earlier, the structure of the config variable is as follows:
+
+```python
+RDM_RECORDS_CUSTOM_FIELDS_UI = [{
+    "section": _("CERN Fields"),
+    "fields": [{
+        "field": "experiment",  # this should be validated against the defined fields in `RDM_RECORDS_CUSTOM_FIELDS`
+        "ui_widget": "MyCustomField",  # custom widget defined in the instance assets
+        "label": _('CERN Experiment'),
+        "placeholder": "Type an experiment...",
+        "icon": "pencil",
+        "description": _(
+            "You should fill this field with one of the experiments e.g LHC, ATLAS etc.",
+        )
+    }]
+}]
+```
+
+The instance administrator is defining a list of sections, with each section defining its own fields respectively. By using these sections, they can semantically group together different fields. The order in which sections and fields are configured is respected in the UI. All the sections and fields will be appended be default in the end of the configured form i.e upload form and community profile page.
+
+Each field can be configured with following parameters
+
+| Config property |  Type  |                          Description                          |
+| :-------------: | :----: | :-----------------------------------------------------------: |
+|      field      | string |      Should be set to one of the defined custom fields.       |
+|    ui_widget    | string |                 Name of the UI widget to use.                 |
+|      label      | string |              UI label to display for the field.               |
+|   placeholder   | string |                  Placeholder for user input.                  |
+|      icon       | string |                    Semantic UI icon name.                     |
+|   description   | string | Description to be used as helptext for the new defined field. |
+
+Moreover, any other property configured in the fields will be used as properties to initialized the respective UI widget.
+
+#### Adding or overriding UI wigets
+
+In the previous section, the `ui_widget` field was mentioned. Several of them
+are already available with InvenioRDM, however, in some cases it might be
+desired to change their behaviour (override) or use a totally new widget.
+
+The webpack loading order for the custom fields is the following:
```

Comment:
Maybe too technical?
```suggestion
UI Widget are discovered and imported in the following way:
```

What if we add a small sentence below explaining the technical part, e.g. Internally, the `webpack` configuration...